plugins/kotlin/j2k/k1.new/src/org/jetbrains/kotlin/nj2k/printing/JKCodeBuilder.kt

```diff
@@ -67,6 +69,9 @@ internal class JKCodeBuilder(context: NewJ2kConverterContext) {
                 (it is JKClass && it.isInterface()) ||
                         (it is JKDeclaration && it.parentOfType<JKClass>()?.isInterface() == true)
             }
+            val isOuterObject = modifiersListOwner is JKClass && modifiersListOwner.classKind == OBJECT && modifiersListOwner.parents()
```

Comment:
Observation: this function is becoming complicated. I think we shouldn't have complicated logic in the code printer. Maybe it makes more sense to move this whole redundancy check to a conversion (for example, to an existing `RemoveWrongExtraModifiersForSingleFunctionsConversion`).